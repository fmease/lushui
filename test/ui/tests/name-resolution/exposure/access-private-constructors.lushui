;;; TEST fail check

module separation of
    data Private: Type of
        private: Private
    
    @abstract
    data Private-Abstract: Type of
        private: Private-Abstract

    @public @abstract
    data Public-Abstract: Type of
        private: Public-Abstract
    
    @(public capsule)
    module further-separation of
        @(public super)
        data Hidden: Type of
            hidden: Hidden
        
        @(public capsule) @abstract
        data Abstract: Type of
            hidden: Abstract

use separation.Private.private as p0
use separation.Private-Abstract.private as p1
use separation.Public-Abstract.private as p2

p3: separation.Public-Abstract = separation.Public-Abstract.private

use separation.further-separation.((Hidden.hidden as h) (Abstract.hidden as ho))
