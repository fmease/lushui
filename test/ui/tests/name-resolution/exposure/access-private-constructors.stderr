error[E029]: binding `topmost.separation.Private` is private
  --> ${DIRECTORY}/name-resolution/exposure/access-private-constructors.lushui:25:16
   |
25 | use separation.Private.private as p0
   |                ^^^^^^^ 
   |

error[E029]: binding `topmost.separation.Private-Abstract` is private
  --> ${DIRECTORY}/name-resolution/exposure/access-private-constructors.lushui:26:16
   |
26 | use separation.Private-Abstract.private as p1
   |                ^^^^^^^^^^^^^^^^ 
   |

error[E029]: binding `topmost.separation.Public-Abstract.private` is private
  --> ${DIRECTORY}/name-resolution/exposure/access-private-constructors.lushui:27:32
   |
27 | use separation.Public-Abstract.private as p2
   |                                ^^^^^^^ 
   |

error[E029]: binding `topmost.separation.Public-Abstract.private` is private
  --> ${DIRECTORY}/name-resolution/exposure/access-private-constructors.lushui:29:61
   |
29 | p3: separation.Public-Abstract = separation.Public-Abstract.private
   |                                                             ^^^^^^^ 
   |

error[E029]: binding `topmost.separation.further-separation.Hidden` is private
  --> ${DIRECTORY}/name-resolution/exposure/access-private-constructors.lushui:31:37
   |
31 | use separation.further-separation.((Hidden.hidden as h) (Abstract.hidden as ho))
   |                                     ^^^^^^ 
   |

error[E029]: binding `topmost.separation.further-separation.Abstract.hidden` is private
  --> ${DIRECTORY}/name-resolution/exposure/access-private-constructors.lushui:31:67
   |
31 | use separation.further-separation.((Hidden.hidden as h) (Abstract.hidden as ho))
   |                                                                   ^^^^^^ 
   |

error: aborting due to 6 previous errors
 note: the error E029 has a detailed explanation
 help: run `lushui explain E029` to view it

